pkg/util/key.go

```diff
@@ -58,11 +58,21 @@ func NewRSAKey() (*KeyBundle, error) {
 }
 
 func WriteFile(path string, b []byte, perm os.FileMode) error {
-	if err := os.MkdirAll(filepath.Dir(path), 0744); err != nil {
+	if err := os.MkdirAll(filepath.Dir(path), os.FileMode(0755)|os.ModeSticky); err != nil {
 		return err
 	}
 
-	return ioutil.WriteFile(path, b, perm)
+	// We need to ensure unix permissions if the file already exists
```

Comment:
Should this read "if the directory already exists"? 

And why? Won't it always be the CSI driver that created these directories.

Reply:
@JoshVanL Maybe ignore that ^^ remark....except to say that I would still have expected the creation of the directory to always have been created in a separate step, by us, with particular permissions, not as a side-effect of writing the key file.
If that's not the case, maybe add an explanatory comment either here or where we call this function, explaining why we need the directory to be world-readable and world-executable.